Rubberduck.RibbonDispatcher/Concrete/RibbonToggleButton.cs

```diff
@@ -0,0 +1,84 @@
+﻿////////////////////////////////////////////////////////////////////////////////////////////////////
+//                                Copyright (c) 2017 Pieter Geerkens                              //
+////////////////////////////////////////////////////////////////////////////////////////////////////
+using System;
+using System.Resources;
+using System.Runtime.InteropServices;
+using System.Diagnostics.CodeAnalysis;
+using stdole;
+
+using Rubberduck.RibbonDispatcher.Abstract;
+using Rubberduck.RibbonDispatcher.AbstractCOM;
+using Rubberduck.RibbonDispatcher.EventHandlers;
+
+namespace Rubberduck.RibbonDispatcher.Concrete {
+    /// <summary>The ViewModel for Ribbon ToggleButton objects.</summary>
+    [SuppressMessage("Microsoft.Interoperability", "CA1409:ComVisibleTypesShouldBeCreatable",
+       Justification = "Publc, Non-Creatable class with exported Events.")]
+    [Serializable]
+    [CLSCompliant(true)]
+    [ComVisible(true)]
+    [ClassInterface(ClassInterfaceType.None)]
+    [ComSourceInterfaces(typeof(IToggledEvents))]
+    [ComDefaultInterface(typeof(IRibbonToggleButton))]
+    [Guid(RubberduckGuid.RibbonToggleButton)]
+    public class RibbonToggleButton : RibbonCommon, IRibbonToggleButton {
+        internal RibbonToggleButton(string id, ResourceManager mgr, bool visible, bool enabled, RdControlSize size,
+                string imageMso, bool showImage, bool showLabel, ToggledEventHandler onToggledAction)
+            : base(id, mgr, visible, enabled, size) {
+            _image     = new ImageObject(imageMso);
+            _showImage = showImage;
+            _showLabel = showLabel;
+            if (onToggledAction != null) Toggled += onToggledAction;
+        }
+        internal RibbonToggleButton(string id, ResourceManager mgr, bool visible, bool enabled, RdControlSize size,
+                IPictureDisp image, bool showImage, bool showLabel, ToggledEventHandler onToggledAction)
+            : base(id, mgr, visible, enabled, size) {
+            _image     = new ImageObject(image);
+            _showImage = showImage;
+            _showLabel = showLabel;
+            if (onToggledAction != null) Toggled += onToggledAction;
+        }
+
+        /// <summary>TODO</summary>
+        public event ToggledEventHandler Toggled;
+
+        /// <summary>TODO</summary>
+        public void OnActionToggle(bool isPressed) {
+            IsPressed = isPressed;
+            Toggled?.Invoke(this, new ToggledEventArgs(isPressed));
+            OnChanged();
+        }
+
+        /// <summary>TODO</summary>
+        public bool       IsPressed { get; private set; }
+        /// <summary>TODO</summary>
+        public new string Label       => IsPressed && ! String.IsNullOrEmpty(LanguageStrings?.AlternateLabel ?? "")
```

Comment:
shouldn't need the null-coalescing operator, since you use `IsNullOrEmpty` already